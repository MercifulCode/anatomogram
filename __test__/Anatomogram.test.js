import React from 'react'
import Enzyme from 'enzyme'
import {shallow} from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

Enzyme.configure({ adapter: new Adapter() })

import Anatomogram from '../src/Anatomogram'

describe(`Anatomogram`, () => {

  test(`should render for a supported species`, () => {
    expect(shallow(<Anatomogram species={`tabula_muris`} />)).toMatchSnapshot();
  })

  test(`should not render for an unsupported species`, () => {
    expect(shallow(<Anatomogram atlasUrl={''} species={`ovis_aries`} />).children()).toHaveLength(0)
  })

})
