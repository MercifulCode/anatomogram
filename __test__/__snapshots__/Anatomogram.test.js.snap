// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Anatomogram should render for a supported species 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Anatomogram
    atlasUrl="https://www.ebi.ac.uk/gxa/"
    species="tabula_muris"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): false,
  Symbol(enzyme.__nodes__): Array [
    false,
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
